File: interpreter/src/main/scala/org/enso/interpreter/nodes/util/SourceLoc.scala
package org.enso.interpreter.nodes.util

case class SourceLoc(position: Long, span: Long) {

  def +(that: SourceLoc): SourceLoc =
    SourceLoc(this.position + that.position, this.span + that.span)
}

object SourceLoc {
  def empty() = SourceLoc(position = 0, span = 0)
  def isEmpty(loc: SourceLoc): Boolean = loc.position == 0 && loc.span == 0
}

File: interpreter/src/main/java/org/enso/interpreter/node/util/SourceLoc.java
package org.enso.interpreter.node.util;

public class SourceLoc {
  public long position = 0;
  public long span = 0;

  public SourceLoc() {}

  public SourceLoc(long position, long span) {
    this.position = position;
    this.span = span;
  }

  public static SourceLoc empty() {
    return new SourceLoc();
  }

  public void add(SourceLoc that) {
    this.position += that.position;
    this.span += that.span;
  }

  public boolean isEmpty() {
    return this.position == 0 && this.span == 0;
  }
}
